fontbe/src/features/marks.rs

```diff
@@ -1069,6 +1334,192 @@ mod tests {
+            tildecomb @(x: 0, y: 742)
+              @(x: 0, y: 578) [acutecomb,brevecomb,macroncomb,tildecomb]
+
+            "#
+        );
+    }
+
+    #[test]
+    fn abvm_blwm_features() {
+        let mut input = MarksInput::default();
+        let out = input
+            .add_glyph("dottedCircle", None, |anchors| {
+                anchors
+                    .add("top", [(297, 552)])
+                    .add("topright", [(491, 458)])
+                    .add("bottom", [(297, 0)]);
+            })
+            .add_glyph("nukta-kannada", None, |anchors| {
+                anchors.add("_bottom", [(0, 0)]);
+            })
+            .add_glyph("candrabindu-kannada", None, |anchors| {
+                anchors.add("_top", [(0, 547)]);
+            })
+            .add_glyph("halant-kannada", None, |anchors| {
+                anchors.add("_topright", [(-456, 460)]);
+            })
+            .add_glyph("ka-kannada", None, |anchors| {
+                anchors.add("bottom", [(290, 0)]);
+            })
+            .add_glyph("ka-kannada.base", None, |anchors| {
+                anchors
+                    .add("top", [(291, 547)])
+                    .add("topright", [(391, 460)])
+                    .add("bottom", [(290, 0)]);
+            })
+            .set_user_fea(
+                "
+            languagesystem DFLT dflt;
+            languagesystem knda dflt;
+            languagesystem knd2 dflt;
+
+            feature psts {
+                sub ka-kannada' halant-kannada by ka-kannada.base;
+            } psts;",
+            )
+            .get_normalized_output();
+
+        assert_eq_ignoring_ws!(
+            out,
+            r#"
+                # abvm: DFLT/dflt, knd2/dflt, knda/dflt ## 2 MarkToBase rules
+                # lookupflag LookupFlag(0)
+                ka-kannada.base @(x: 291, y: 547)
+                  @(x: 0, y: 547) candrabindu-kannada
+                ka-kannada.base @(x: 391, y: 460)
+                  @(x: -456, y: 460) halant-kannada
+
+                # blwm: DFLT/dflt, knd2/dflt, knda/dflt ## 2 MarkToBase rules
+                # lookupflag LookupFlag(0)
+                ka-kannada @(x: 290, y: 0)
+                  @(x: 0, y: 0) nukta-kannada
+                ka-kannada.base @(x: 290, y: 0)
+                  @(x: 0, y: 0) nukta-kannada
+
+                # mark: DFLT/dflt, knd2/dflt,knda/dflt ## 3 MarkToBase rules
+                # lookupflag LookupFlag(0)
+                dottedCircle @(x: 297, y: 0)
+                  @(x: 0, y: 0) nukta-kannada
+                dottedCircle @(x: 297, y: 552)
+                  @(x: 0, y: 547) candrabindu-kannada
+                dottedCircle @(x: 491, y: 458)
+                  @(x: -456, y: 460) halant-kannada
+
+                "#
+        );
+    }
+
+    #[test]
+    fn mark_on_gsub_glyph() {
+        let mut out = MarksInput::default();
+        let out = out
+            // a.alt is only reachable via substitution
```

Comment:
I don't think this is testing the gsub closure, the same mark feature would be generated even without user FEA snippet sub a by a.alt below. Maybe you wanted to test `abvm_on_gsub_glyph` instead?

Reply:
I need to explain this a bit better; this is checking is that if we _don't_ have any abvm scripts, we put all glyphs into the 'non abvm' bucket, even if they are only reachable via substitution. (This exists because I had a bug where I wasn't handing this correctly)

(will add a better explanation)